src/components/editor/editor-pane/tool-bar/table-picker/table-picker.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-FileCopyrightText: 2020 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { useEffect, useRef, useState } from 'react'
+import { Button } from 'react-bootstrap'
+import { useTranslation } from 'react-i18next'
+import { useClickAway } from 'react-use'
+import { ForkAwesomeIcon } from '../../../../common/fork-awesome/fork-awesome-icon'
+import { TablePickerDialog } from './table-picker-dialog'
+import './table-picker.scss'
+
+export interface TablePickerProps {
+  show: boolean
+  onDismiss: () => void
+  onTablePicked: (row: number, cols: number) => void
+}
+
+export type TableSize = {
+  rows: number,
+  columns: number
+}
+
+export const TablePicker: React.FC<TablePickerProps> = ({ show, onDismiss, onTablePicked }) => {
+  const { t } = useTranslation()
+  const containerRef = useRef<HTMLDivElement>(null)
+  const [tableSize, setTableSize] = useState<TableSize>()
+  const [showDialog, setShowDialog] = useState(false)
+
+  useClickAway(containerRef, () => {
+    onDismiss()
+  })
+
+  useEffect(() => {
+    setTableSize(undefined)
+  }, [show])
+
+  return (
+    <div className={`position-absolute table-picker-container p-2 ${!show ? 'd-none' : ''} bg-light`} ref={containerRef} role="grid">
+      <p className={'lead'}>
+        { tableSize
```

Comment:
If you use a const initial value that you use in the `useState` and the effect, then you don't have to handle the `undefined` here.

Reply:
This is undesired in this case as it would complicate more lines than it simplifies…